### PR TITLE
Feat(trends):t_score function

### PR DESCRIPTION
**Pull request by wmak**
- This adds a [t score](https://en.wikipedia.org/wiki/Welch%27s_t-test) function to help us identify significant trends
- This introduces variance as a range function as well, to support the calculation of the t score
- Adding a default t_score filter, we'll eventually have the ability for the user to change the number as "low/med/high" but for now filtering >1 to remove the ones we have no confidence in.

### CODE REVIEW

File: src/sentry/api/endpoints/organization_events_trends.py [hunk 1]
```diff
@@ -39,13 +40,39 @@ class OrganizationEventsTrendsEndpointBase(OrganizationEventsV2EndpointBase):
             "format": "user_misery_range({}, {start}, {end}, {index})",
             "alias": "user_misery_range_",
         },
+        "variance": {
+            "format": "variance_range(transaction.duration, {start}, {end}, {index})",
```
[the remark this answers is not in the document]
Comment (wmak) [reply src/sentry/api/endpoints/organization_events_trends.py:44]: 👍 I think eventually all of these will need to be updated to support measurements.

**Comment by trillville** [comment]
Oh shoot - one thing I forgot to mention in our meeting today: t-score calculations are very sensitive to outliers, would it make sense/be feasible to be a bit more aggressive with the default transaction maximum length filters?

File: src/sentry/static/sentry/app/views/performance/trends/utils.tsx [hunk 2]
```diff
@@ -381,14 +379,10 @@ export function movingAverage(data, index, size) {
...
-  trendFunction: TrendFunction,
-  trendChangeType: TrendChangeType
-) {
-  const aliasedPercentage = getTrendAliasedQueryPercentage(trendFunction.alias);
-  let limitQuery = aliasedPercentage + ':<1';
+function getLimitTransactionItems(trendChangeType: TrendChangeType) {
+  let limitQuery = 'trend_percentage():<1 t_score():>1';
   if (trendChangeType === TrendChangeType.REGRESSION) {
-    limitQuery = aliasedPercentage + ':>1';
+    limitQuery = 'trend_percentage():>1 t_score():<-1';
```
Comment (trillville) [inline src/sentry/static/sentry/app/views/performance/trends/utils.tsx:385]: would personally vote for +/- 2 as the default cutoff, +/- 1 corresponds to a super low 70% confidence level (http://uregina.ca/~gingrich/tt.pdf)

Comment (wmak) [reply src/sentry/static/sentry/app/views/performance/trends/utils.tsx:385]: :+1: There's an upcoming PR to replace this with the Low/Medium/High mentioned

**Comment by wmak** [comment]
:+1: I'll update the `transaction.duration:<=60min` to `transaction.duration:<=15min`

File: tests/snuba/api/endpoints/test_organization_events_trends.py [hunk 3]
```diff
@@ -537,6 +563,8 @@ def test_misery_trend_function(self):
             "minus_user_misery_range_2_user_misery_range_1": 1.0,
             "percentage_user_misery_range_2_user_misery_range_1": 2.0,
         }
+        for key, value in expected_data.items():
+            assert events["data"][0][key] == value
```
Comment (markstory) [inline tests/snuba/api/endpoints/test_organization_events_trends.py:567]: You might want to consider a helper for this like `assert_event(events["data"][0], expected_data)`

**Review by markstory — APPROVED** [review]

**Pull request merged**